ApplicationCode/GeoMech/GeoMechDataModel/RigFemPartResultCalculatorMudWeightWindow.cpp

```diff
@@ -270,7 +271,7 @@ RigFemScalarResultFrames* RigFemPartResultCalculatorMudWeightWindow::calculate(
                         // Initial overburden gradient
                         if ( OBG0FromGrid )
                         {
-                            OBG0 = obg0FrameData[nodeIdx];
+                            OBG0 = obg0FrameData[elmNodResIdx];
```

Comment:
Consider moving all methods using the nodeIdx into a separate (sub) scope to avoid index mixing. This will also make it clear where to work with node based structures.